.github/workflows/ci.yml

```diff
@@ -3,6 +3,33 @@ name: CI
 on: [push]
 
 jobs:
+  clang_check:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set-up repository
+      uses: actions/checkout@v2
+
+    - name: Install clang-format
+      run: |
+        sudo apt install -y clang-format-8
+
+    - name: Run clang formatter
+      run: |
+        ALL_FILES_TO_CHECK=$(find . -type f -regex ".*\.\(cpp\|hpp\|c\|h\)" -print | paste -sd " ")
+        for file in ${ALL_FILES_TO_CHECK}; do
+          clang-format-8 -i -verbose ${file}
+        done
+        changes="$(git diff)"
+        if [[ ! -z ${changes} ]]; then
```

Comment:
The `$changes` should be in double quotes here to supply the value as a single argument to the `test` command:
`[[ ! -z "$changes" ]]`, also you can use just `[[ "$changes" ]]`.
If you used `[ ]` instead of `[[ ]]` the unquoted version wouldn't work.
Generally speaking, if you have a variable that is expected to have some whitespace in it, quote it just to be safe.